### PR TITLE
feat: download, install and run DLKcat from MATLAB

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
`runDLKcat` can download the DLKcat package, install it and its requirements (in a virtualenv with pip) and run this Python package via MATLAB. By default, DLKcat is installed in `GECKO/dlkcat`.

Tested on Windows (requires WSL), not yet tested on Mac/Linux.

For Windows, you can ensure that pip and pipenv are installed by running: `sudo apt install python3-pip && pip install pipenv` in your WSL terminal.

**I hereby confirm that I have:**

- [x] Tested my code with [all requirements](https://github.com/SysBioChalmers/GECKO) for running GECKO
- [x] Selected `devel3` as a target branch (top left drop-down menu)

### CODE REVIEW

File: src/geckomat/gather_kcats/runDLKcat.m [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    WSLpath.Output = getWSLpath(DLKcatOutput);
+    WSLcommand = ['wsl --shell-type login --cd ' WSLpath.DLKcat '/'];
+    fprintf(['=== Below, status messages from WSL are shown.\n'...
+             '=== Warnings about bogus screen size can safely be ignored.\n'...
+             '=== Preparing DLKcat environment...\n\n'])    
+    system([WSLcommand ' pipenv install -r requirements.txt'],'-echo');
+    fprintf('\n=== Running DLKcat prediction, this may take several minutes...\n\n')    
+    system([WSLcommand ' pipenv run python3 DLKcat.py ' WSLpath.Input ' ' WSLpath.Output],'-echo');
+else
+    [~,pythonCheck] = system('python3 --version');
```
Comment (feiranl) [inline src/geckomat/gather_kcats/runDLKcat.m:77]: Could also add a line to check the existence of pipenv. 
`pipenv --version`
I am not sure even though I installed pipenv in the base and can successfully `run pipenv install -r requirements.txt` in terminal, but failed from matlab.

Comment (edkerk) [reply src/geckomat/gather_kcats/runDLKcat.m:77]: Is that on Mac?

Comment (ae-tafur) [reply src/geckomat/gather_kcats/runDLKcat.m:77]: Same for me on Mac

Comment (edkerk) [reply src/geckomat/gather_kcats/runDLKcat.m:77]: Thanks for the feedback. I'll change this PR to draft while I try to fix it.

**Comment by edkerk** [comment]
Now significantly refactored (no WSL in Windows) and tries to catch potential issues with e.g. name of python binary (python/python3), and whether pip packages are on the system path.

Works on Windows with Python 3.9.10. In case of problems, it could be helpful to remove the current pipenv in the GECKO/dlkcat folder by running `pipenv --rm` in the operating system's terminal.

**Review by johan-gson — APPROVED** [review]
I tested it and it works, I didn't look through it in super detail.

File: src/geckomat/gather_kcats/runDLKcat.m [hunk 2]
```diff
@@ -59,96 +59,79 @@ function runDLKcat(DLKcatInput, DLKcatOutput, modelAdapter, DLKcatPath)
...
-
-if ispc
-    % Python
-    [checks.python.status, checks.python.out] = system('python --version');
-    if ~startsWith(checks.python.out,"Python 3.")
+% Python
+three = ''; %suffix (python vs. python3)
+[checks.python.status, checks.python.out] = system('python --version');
+if checks.python.status ~= 0
+    [checks.python.status, checks.python.out] = system('python3 --version');
```
Comment (ae-tafur) [inline src/geckomat/gather_kcats/runDLKcat.m:66]: check python3 --version returns status 0 on Mac

**Comment by ae-tafur** [comment]
works on Mac

**Review by ae-tafur — APPROVED** [review]

**Comment by edkerk** [comment]
Works here on:
- Windows 10, Python 3.9.13, installed via Python.org (not via Windows Store)
- Mac 11.7.2, Python 3.9.13, installed via Python.org (not via Xcode)

**Review by feiranl — APPROVED** [review]
Works fine on Mac.

**Comment by edkerk** [comment]
Does **not** work on Python 3.10.x, as biopython seems not available as binary and is instead trying to compile it.
I'll make runDLKcat to have an optional "pythonPath" variable that refers to which python binary should be used. To give it a bit more flexibility than to rely on what `PATH` finds.

**Pull request merged**